join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
 
-##! join__Graph
+Each supergraph schema contains a list of the subgraphs. It represents this as an enum called [{join__Graph}](#join__Graph) with an enum value for each subgraph. Each enum value is annotated with a [{@join__graph}](#@join__graph) directive telling the router what endpoint can be used to reach the subgraph and a name for the subgraph that can be used in representations of query plans and diagnostic messages.
 
-Enumerate subgraphs.
+To resolve a field, the router needs to know to which subgraphs it can delegate the field's resolution. The most direct way to indicate this in a supergraph schema is by annotating the field with a [{@join__field}](#@join__field) directive specifying which subgraph should be used to resolve that field.
 
-```graphql definition
-enum join__Graph
+In order for the router to send an operation that resolves a given field on a parent object to a subgraph, the operation needs to first resolve the parent object itself. There are several ways to accomplish this, described below. The examples below include abbreviated versions of the supergraph schemas which do not include the `schema` definition, directive definitions, or the `join__Graph` definition. This specification does not require the subgraph queries to be the same as those described in these examples; this is just intended to broadly describe the meanings of the directives.
+
+## Root fields
+
+If this field appears at the root of the overall operation (query or mutation), then the field can just be placed at the root of the subgraph operation.
+
+```graphql example -- Root fields
+# Supergraph schema
+type Query {
+  fieldA: String @join__field(graph: A)
+  fieldB: String @join__field(graph: B)
+}
+
+# Operation
+{ fieldA fieldB }
+# Generated subgraph operations
+## On A:
+{ fieldA }
+## On B:
+{ fieldB }
 ```
 
-Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
-:::[example](photos.graphql#join__Graph) -- Using join__Graph to define endpoints
+## Fields on the same subgraph as the parent operation
 
-The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
+If this field's parent field will be resolved by an operation on the same subgraph, then this field can be resolved as part of the same operation, by putting it in a nested selection set on the parent field's subgraph operation. Note that this example contains a {@join__owner} directive on an object type; this will be described later.
```

Comment:
"this" again feels strange to me
```suggestion
If a field's parent field will be resolved by an operation on the same subgraph, then it can be resolved as part of the same operation, by putting it in a nested selection set on the parent field's subgraph operation. Note that this example contains a {@join__owner} directive on an object type; this will be described later.
```

Reply:
heh, already applied in aab84d4